src/models/stores/documents.ts

```diff
@@ -27,120 +27,111 @@ export const DocumentsModel = types
       });
     }
   }))
-  .views(self => ({
-    getNextPersonalDocumentTitle(user: UserModelType, base: string) {
-      let maxUntitled = 0;
-      self.byTypeForUser(PersonalDocument, user.id)
-        .forEach(document => {
-          const match = /.*-([0-9]+)$/.exec(document.title || "");
-          // length check to skip timestamps
-          if (match && match[1] && (match[1].length < 4)) {
-            const suffix = parseInt(match[1], 10);
-            maxUntitled = Math.max(maxUntitled, suffix);
-          }
-        });
-      return `${base}-${++maxUntitled}`;
-    },
-
-    getPersonalDocument(userId: string) {
-      return self.all.find((document) => {
-        return (document.type === PersonalDocument) && (document.uid === userId);
-      });
-    },
-
-    getLearningLogDocument(userId: string) {
-      return self.all.find((document) => {
-        return (document.type === LearningLogDocument) && (document.uid === userId);
-      });
-    },
-
-    getProblemDocument(userId: string) {
-      return self.all.find((document) => {
-        return (document.type === ProblemDocument) && (document.uid === userId);
-      });
-    },
-
-    getProblemDocumentsForGroup(groupId: string) {
-      return self.all.filter((document) => {
-        return (document.type === ProblemDocument) && (document.groupId === groupId);
-      });
-    },
-
-    getLastPublishedProblemDocumentsForGroup(groupId: string) {
-      return self.all.filter((document) => {
-        return (document.type === ProblemPublication) && (document.groupId === groupId);
+  .views(self => {
+    const getLatestPublications = (publications: DocumentModelType[], attr: "uid" | "originDoc") => {
```

Comment:
FYI, simply moving this function to the previous `views` block would obviate the need for the additional closure here. On the one hand, it could be considered an implementation detail which would be an argument for encapsulating it this way. On the other hand, it might be useful to unit-test it separately (at some point), which would be an argument for making it part of the public API.

Reply:
@kswenson  It felt more like a helper function so that is why I didn't add it as a view.  I just pushed a commit to make the diff more readable by pulling that helper function out of the model declaration since it doesn't depend on the state.  You might want to take a look again since the changes are more evident now.